pkg/router/routeapihelpers/validation.go

```diff
@@ -281,6 +290,27 @@ func ExtendedValidateRoute(route *routev1.Route) field.ErrorList {
 	return result
 }
 
+// validatePEMContent validates that the passed content, when PEM decoded contains
+// the expected type. This validation is necessary to block attempts of passing certificates
+// inside the field that should be private key only, and vice versa. This is required
+// because otherwise passing an out of order certificate to HAProxy will break it.
+// The function will take the array of bytes of a key/cert, and the expected type
+// that the PEM should content. The match here is not strict, but will check if the
+// passed byte type CONTAINS the expected pemType
```

Comment:
I'm being picky about this, but the comment here isn't clearly describing what's going on.  Let me know if you think this works.  If I've misunderstood, we may need to change the code.

```suggestion
// validatePEMContent takes content and pemType, PEM decodes content and 
// validates that the first block matches the expected pemType. This validation
// 1. Ensures that the required pemType is first in the order
// 2. Blocks attempts of passing certificates where keys should be used, and
// 3. Blocks attempts of passing keys where certificates should be used.
// Passing an out of order certificate, or key as certificate, or certificate as key
// breaks HAProxy.  Note that the match of content and pemType is not exact, but
// content must CONTAIN the expected pemType.
```